src/psyclone/tests/dynamo0p3_builtins_test.py

```diff
@@ -1656,6 +1656,53 @@ def test_inc_X_powreal_a():  # pylint: disable=invalid-name
             assert output in code
 
 
+def test_inc_X_powint_n():  # pylint: disable=invalid-name
+    ''' Test that 1) the str method of DynIncXPowintNKern returns the
+    expected string and 2) we generate correct code for the built-in
+    operation X = X**n where 'n' is an integer scalar and X is a field '''
+    for distmem in [False, True]:
+        _, invoke_info = parse(
+            os.path.join(BASE_PATH,
+                         "15.6.2_inc_X_powint_n_builtin.f90"),
+            distributed_memory=distmem,
+            api="dynamo0.3")
+        psy = PSyFactory("dynamo0.3",
+                         distributed_memory=distmem).create(invoke_info)
+        # Test string method
+        first_invoke = psy.invokes.invoke_list[0]
+        kern = first_invoke.schedule.children[0].children[0]
+        assert str(kern) == "Built-in: raise a field to an integer power"
+        # Test code generation
+        code = str(psy.gen)
+        print code
+        if not distmem:
+            output = (
+                "      ndf_any_space_1_f1 = f1_proxy%vspace%get_ndf()\n"
+                "      undf_any_space_1_f1 = f1_proxy%vspace%get_undf()\n"
+                "      !\n"
+                "      ! Call our kernels\n"
+                "      !\n"
+                "      DO df=1,undf_any_space_1_f1\n"
+                "        f1_proxy%data(df) = f1_proxy%data(df)**i_scalar\n"
+                "      END DO \n"
+                "      !\n")
+        else:
+            mesh_code_present("f1", code)
+            output = (
+                "      ! Call kernels and communication routines\n"
+                "      !\n"
+                "      DO df=1,f1_proxy%vspace%get_last_dof_owned()\n"
+                "        f1_proxy%data(df) = f1_proxy%data(df)**i_scalar\n"
+                "      END DO \n"
+                "      !\n"
+                "      ! Set halos dirty for fields modified in the above "
+                "loop\n"
+                "      !\n"
+                "      CALL f1_proxy%set_dirty()")
+
```

Comment:
New test for new built-in - good.

Reply:
Please could you add a check (to this test) that the generated code compiles. Look for 'compile' elsewhere in this file or perhaps in `dynamo0p3_test.py`.

Reply:
Added the check and happy to add checks for the other tests in the file as well as part of this issue. 
Just a warning that for me the compile fails with errors such as cited below.

   integer, parameter :: r_def     = real64
                                          1
   Error: Symbol 'real64' at (1) has no IMPLICIT type
/net/home/h06/ikavcic/PSyclone_work/PSyclone/src/psyclone/tests/test_files/dynamo0p3/infrastructure/constants_mod.f90:55.42:

   integer, parameter :: r_single  = real32
                                          1
   Error: Symbol 'real32' at (1) has no IMPLICIT type
/net/home/h06/ikavcic/PSyclone_work/PSyclone/src/psyclone/tests/test_files/dynamo0p3/infrastructure/constants_mod.f90:56.42:

   integer, parameter :: r_double  = real64
                                          1
   Error: Symbol 'real64' at (1) has no IMPLICIT type
/net/home/h06/ikavcic/PSyclone_work/PSyclone/src/psyclone/tests/test_files/dynamo0p3/infrastructure/constants_mod.f90:62.41:

   integer, parameter :: i_def     = int32

   Error: Symbol 'int32' at (1) has no IMPLICIT type
/net/home/h06/ikavcic/PSyclone_work/PSyclone/src/psyclone/tests/test_files/dynamo0p3/infrastructure/constants_mod.f90:80.17:

Reply:
I think it's best if we only touch the tests that you've already touched - small changesets are best!
If the compilation fails then this needs to be fixed as part of this ticket. Those symbols that the compiler is complaining about are from the iso_c_env module. Which compiler are you using?

Reply:
Compilation report for "pytest --compile src/psyclone/tests/dynamo0p3_builtins_test.py" on master for different compilers we have on Desktop (branch behaves the same).

- GNU 5.3.0, 6.1.0 and 7.1.0: everything works.
- Intel 15.0.0, 16.0.1, 17.0.1: fails with the above error.
- Portland 15.7, 16.10: fails with the above error.

There should be a good support for iso_c_env module in all languages. I'll make some more checks and report back.

Reply:
I just talked with Matthew, he thinks it's because compiler flags are set specifically for GNU. I'm checking that.

Reply:
SOLVED: As the default is `gfortran`, all it needed was to explicitly specify `--f90="compiler name"`. Eg:

`py.test --compile --f90="pgfortran" src/psyclone/tests/dynamo0p3_builtins_test.py` for Portland

`py.test --compile --f90="ifort" src/psyclone/tests/dynamo0p3_builtins_test.py` for Intel

It is a bit slower than Gnu but it works :)

Reply:
Hurrah :-) Well done for finding the `--f90` flag and apologies for not mentioning it before.